docs/DeveloperGuide.md

```diff
@@ -433,21 +433,54 @@ Key methods in this implementation include:
+
+Most validation is handled by the `Balance` class, through three static methods:
+- `isValidDollarString(String test)`: Checks if a given string is a valid human-friendly string representation of a balance using a validation regex.
+- `isWithinBalanceLimit(Integer balanceInCents)`: Checks if a given integer is a valid balance value that is a field of a contact. CampusConnect restricts users to only be able to track a maximum payment owed to / from a contact of $10,000.
+- `isWithinTransactionLimit(Integer amountInCents)`: Checks if a given integer is a valid money value that is used in a single transaction. Due to the $10,000 balance limit, any individual transaction cannot transact more than $20,000, since that will always lead to the balance limit being exceeded. This check occurs whenever a `Balance` object is instantiated, since there should never be any `Balance` objects with a value greater than $20,000.
+
+The `PayCommand` and `OweCommand` classes are extremely similar. Both classes simply update the Person's balance with the amount given.
+
+The following activity diagram shows the logic of the validation checks that occur when a user uses the `pay` command:
+
+<puml src="diagrams/PayActivityDiagram.puml" alt="PayActivityDiagram" />
+
+The series of validation checks in the activity diagram also applies to the `owe` command.
 
 #### Design considerations
 
-**Aspect: Using integers to represent money value**
+**Aspect: Money representation data type**
+
+There are multiple options to represent money values in the application within the Balance class.
 
 * **Alternative 1 (current choice):** Use `int` to represent money value in cents.
-  * **Pros**: Simple and straightforward.
-  * **Cons**: Requires more complex processing when converting to human-friendly string representations.
+  * Pros: Simple and straightforward, and not vulnerable to floating-point precision errors.
+  * Cons: Requires more complex processing when converting to human-friendly string representations.
 * **Alternative 2:** Use `double` to represent money value in dollars.
-  * **Pros**: More human-friendly.
-  * **Cons**: Vulnerable to floating-point precision errors.
+  * Pros: More human-friendly.
+  * Cons: Vulnerable to floating-point precision errors.
+
+**Aspect: Specialization of commands**
+
+The payments feature could be implemented as a single command, with a negative sign to specify whether the payment is a `pay` or `owe` command.
 
+* **Alternative 1 (current choice):** Use two separate `pay` and `owe` commands to represent positive and negative transactions.
+  * Pros: More straightforward to understand by a human by thinking "I pay $X to contact Y" or "I owe $X to contact Y".
```

Comment:
```suggestion
 * Pros: Make the concept more straightforward for a human to understand; humans verbalize their thoughts by saying "I pay $X to contact Y" or "I owe $X to contact Y".
```

the two consecutive `by` makes it sound gramatically incorrect

Reply:
Makes sense!